Zend/Optimizer/zend_optimizer.c

```diff
@@ -974,6 +974,28 @@ zend_function *zend_optimizer_get_called_func(
 				}
 			}
 			break;
+		case ZEND_INIT_PARENT_PROPERTY_HOOK_CALL: {
+			zend_class_entry *scope = op_array->scope;
+			ZEND_ASSERT(scope != NULL);
+			if (scope && (scope->ce_flags & ZEND_ACC_LINKED) && scope->parent) {
```

Comment:
The line above you're asserting scope is not NULL, yet in this line you check whether scope is not NULL. That doesn't really make much sense.